packages/offix-cache/src/cache/mutations.ts

```diff
@@ -75,7 +75,7 @@ export const createMutationOptions = <T = {
     }
   };
   options.update = update;
-  options.context = { ...context, returnType };
+  options.context = { ...context, returnType, idField };
```

Comment:
Would this work only when we use the helper?

Reply:
yes, but `offlineMutation` in the client uses this function. we already say in our docs that if you want to use conflicts you need to provide `returnType` to `offlineMutation` or directly in the context so it would be the same here.

Do you have any other ideas on how to do this?

Reply:
It is on client.cache.config.dataId from object?

https://github.com/apollographql/apollo-client/blob/master/packages/apollo-cache-inmemory/src/inMemoryCache.ts#L204

Reply:
@wtrocki it looks like this function uses __typename from _result_. From my understanding, we won't have access to anything that can give us the id field at this point. The first time we will have access is by using the optimistic response as a result, which uses the `idField` variable anyway.

Correct me if I am wrong

Reply:
Ok.. so we have helpers that do not support many use cases for the moment. 
And then we have id mapping and base feature that relies on the value from helpers. We looking for full outer join for these cases. 

For base logic, we actually have both typename and id (and can use this function under the hood. For id mapping we have access to result that has everything to calculate the id (as this stuff will land in cache seconds after mapping is done).

Reply:
our helpers accept `id` which is typical field and we have cache accessible it witch is `__typename+id` by default but it can be whatever you want - including new Date()

Reply:
Thinking of it telling people - use `id` is probably too much